reader.go

```diff
@@ -0,0 +1,112 @@
+// Copyright (C) 2022 by Posit Software, PBC
+package rsf
+
+import (
+	"bufio"
+	"encoding/binary"
+	"fmt"
+	"io"
+)
+
+type reader struct {
+	pos int
```

Comment:
Should the `reader` include an `io.Reader` in the struct instead of passing it as a parameter in the methods?

Reply:
See my comment above.